ee/fleetd-chrome/src/tables/privacy_preferences.ts

```diff
@@ -0,0 +1,100 @@
+import Table from "./Table";
+
+export default class TablePrivacyPreferences extends Table {
+  // expose properties available from the chrome.privacy API as a virtual osquery "table"
+  // https://developer.chrome.com/docs/extensions/reference/privacy/
+
+  name = "privacy_preferences";
+  propertyAPIs = {
+    // all of type `types.ChromeSetting<boolean>` with default `true` unless otherwise specified
+
+    // though all of these properties are documented, some have not been added to the
+    // typings we are using: https://www.npmjs.com/package/@types/chrome and must be `@ts-ignore`d
+
+    // network
+    network_prediction_enabled: chrome.privacy.network.networkPredictionEnabled,
+    // type `types.ChromeSetting<IPHandlingPolicy>`, default "default"
+    // IPHandlingPolicy:
+    // "default" | "default_public_and_private_interfaces" |
+    // "default_public_interface_only" | "disable_non_proxied_udp"
+    web_rtc_ip_handling_policy: chrome.privacy.network.webRTCIPHandlingPolicy,
+
+    // services
+    autofill_address_enabled: chrome.privacy.services.autofillAddressEnabled,
+    autofill_credit_card_enabled:
+      chrome.privacy.services.autofillCreditCardEnabled,
+    // DEPRECATED and replaced with above two properties
+    autofill_enabled: chrome.privacy.services.autofillEnabled,
+    save_passwords_enabled: chrome.privacy.services.passwordSavingEnabled,
+    safe_browsing_enabled: chrome.privacy.services.safeBrowsingEnabled,
+    // default false
+    safe_browsing_extended_reporting_enabled:
+      chrome.privacy.services.safeBrowsingExtendedReportingEnabled,
+    search_suggest_enabled: chrome.privacy.services.searchSuggestEnabled,
+    // default false
+    spelling_service_enabled: chrome.privacy.services.spellingServiceEnabled,
+    translation_service_enabled:
+      chrome.privacy.services.translationServiceEnabled,
+
+    // websites
+    // @ts-ignore
+    ad_measurement_enabled: chrome.privacy.websites.adMeasurementEnabled,
+    // default false
+    do_not_track_enabled: chrome.privacy.websites.doNotTrackEnabled,
+    // @ts-ignore
+    fledge_enabled: chrome.privacy.websites.fledgeEnabled,
+    hyperlink_auditing_enabled:
+      chrome.privacy.websites.hyperlinkAuditingEnabled,
+    // @ts-ignore, DEPRECATED
+    privacy_sandbox_enabled: chrome.privacy.websites.privacySandboxEnabled,
+    // WINDOWS AND CHROMEOS ONLY
+    protected_content_enabled: chrome.privacy.websites.protectedContentEnabled,
+    referrers_enabled: chrome.privacy.websites.referrersEnabled,
+    third_party_cookies_allowed:
+      chrome.privacy.websites.thirdPartyCookiesAllowed,
+    // @ts-ignore
+    topics_enabled: chrome.privacy.websites.topicsEnabled,
+  };
+
+  columns = Object.keys(this.propertyAPIs);
+
+  async generate() {
```

Comment:
In this table implementation it does make sense to capture error (try/catch) if any of the API fails, as we don't want to fail the whole table run if one API failed. For now, if an API fails we should return the corresponding column with empty value.

Alternative idea: Why not have one table for each property? (maybe that can be implemented with an abstraction). This is something to think about moving forward, not something to block now.

Let me know if it makes sense.

Reply:
Makes perfect sense, sounds good. That's what I was thinking, just wasn't sure how to proceed given that we wanted other errors to bubble up.

We could also store any errors we catch, then pass them into some custom error handling system before returning the table results? For now I'll just log them. 

Thank you!

Reply:
Sounds good.

Forgot to mention the *current* (may change in the future) approach was discussed with Zach+Rachel.

To sum it up for future reference:
- If table column results relies on one API, then do not capture error, let it bubble up so it can reach Fleet.
- If table results consists of many API calls for different columns then let's capture the error and return some of the columns that do have some valid values (and log the errors).